lib/dialects/mysql/query/mysql-querycompiler.js

```diff
@@ -9,12 +9,13 @@ class QueryCompiler_MySQL extends QueryCompiler {
     super(client, builder, formatter);
 
     const { returning } = this.single;
-    if (returning) {
+    if (returning && (!client.version || parseFloat(client.version) < 10.5)) {
```

Comment:
since mysql does not support `returning`, only maria does, we need to check not just version. Probably we can introduce another field alongside `client` and `version` that would indicate specific db type, and include it in the check.

Reply:
You are right, but at the moment it does NOT result in the wrong warning.
Because the latest version of `MySQL` is much smaller than `10.5.0`.

Reply:
It's still hacky and confusing, and should MySQL ever go up to version 10, things will suddenly break. Let's handle it from the very beginning.

Reply:
All right. I suggest putting the `type` field as a new field alongside `client` and `version`.

Reply:
>  I suggest putting the type field as a new field alongside client and version.

Sure, that works.